hs_core/views/utils.py

```diff
@@ -418,21 +419,24 @@ def link_irods_file_to_django(resource, filename, size=0):
             if not ResourceFile.objects.filter(object_id=resource.id,
                                                fed_resource_file_name_or_path=filename).exists():
                 ResourceFile.objects.create(content_object=resource,
-                                            resource_file=None,
+                                            resource_file=None, mime_type=file_format_type,
                                             fed_resource_file_name_or_path=filename,
                                             fed_resource_file_size=size)
                 b_add_file = True
 
         elif not ResourceFile.objects.filter(object_id=resource.id,
                                              resource_file=filename).exists():
-                ResourceFile.objects.create(content_object=resource,
+                ResourceFile.objects.create(content_object=resource, mime_type=file_format_type,
                                             resource_file=filename)
                 b_add_file = True
 
         if b_add_file:
-            file_format_type = get_file_mime_type(filename)
+            # file_format_type = get_file_mime_type(filename)
             if file_format_type not in [mime.value for mime in resource.metadata.formats.all()]:
                 resource.metadata.create_element('format', value=file_format_type)
+            # this should assign a logical file object to this new zip file
+            # if this resource supports logical file
+            resource.set_default_logical_file()
```

Comment:
I also need to change the first comment at the beginning of this function accordingly.